src/StrategyReserves.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IStrategy} from "@interfaces/IStrategy.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title IStrategyWithOwnership
+ * @notice Interface for strategy contracts with ownership functionality
+ * @dev Extends IStrategy with ownership methods from Ownable
+ */
+interface IStrategyWithOwnership is IStrategy {
+    /**
+     * @notice Transfers ownership of the strategy to a new owner
+     * @param newOwner The address to transfer ownership to
+     */
+    function transferOwnership(address newOwner) external;
+
+    function owner() external view returns (address);
+}
+
+/**
+ * @title StrategyReserves
+ * @notice Contract for managing a pool of strategies that users can claim
+ * @dev Maintains a list of strategies owned by the contract that users can claim by depositing tokens
+ */
+contract StrategyReserves is Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @notice The ERC20 token contract
+    IERC20 public immutable token;
+
+    /// @notice Array of available strategies
+    address[] public strategies;
+
+    /// @notice Mapping to track which strategies have been added for O(1) duplicate checking
+    mapping(address => bool) public strategyExists;
+
+    /// @notice Minimum amount required to claim a strategy (in token's native decimals)
+    uint256 public minimumClaimAmount = 1e6; // Default: 1 token (assuming 6 decimals)
+
+    /// @notice Emitted when a strategy is added to the contract
+    event StrategyAdded(address indexed strategy);
+
+    /// @notice Emitted when a strategy is claimed by a user
+    event StrategyClaimed(address indexed user, address indexed strategy, uint256 amount);
+
+    /// @notice Emitted when a strategy is claimed on behalf of an address
+    event StrategyClaimedForAddress(address indexed beneficiary, address indexed strategy);
+
+    /// @notice Emitted when the minimum claim amount is updated
+    event MinimumClaimAmountUpdated(uint256 oldAmount, uint256 newAmount);
+
+    /**
+     * @notice Constructor that initializes the factory with ERC20 token address
+     * @param _token Address of the ERC20 token
+     * @param _owner Address of the factory owner
+     */
+    constructor(address _token, address _owner, uint256 _minimumClaimAmount) Ownable(_owner) {
+        require(_token != address(0), "Invalid token address");
+        token = IERC20(_token);
+        minimumClaimAmount = _minimumClaimAmount;
+    }
+
+    /**
+     * @notice Adds a strategy to the available pool
+     * @dev The factory must be the owner of the strategy when adding. Duplicate strategies are not allowed.
+     * @param strategy The address of the strategy to add
+     */
+    function add(address strategy) external onlyOwner {
```

Comment:
This contract should inherit from the factory contract. Having two contracts results in 3 transactions for the backend: (1. deploy the contract from the factory, 2. transfer ownership to this contract, 3. call add). The `add` function should be invoked from the createStrategy function. This way, a single call can execute all the necessary actions.